### DISCUSSION TITLE
`HANConv` questions: Featurizing other nodes? Symmetries inserted into reverse edges?

### DISCUSSION THREAD

**Discussion by jasperhyp** [pyg-team/pytorch_geometric#7262] [Q&A] [answered]
Hi!  I have a few questions regarding the use of `HANConv` on some large KG (~50K nodes) after reading the original paper and the tutorial in the document.  I am only interested in one node type, i.e. I would only extract the embeddings after message passing for one node type (let's say `A`) for some downstream node classification task.

1. **Is it expected that one would still want to consider passing messages directly from other node types to one another (i.e. keeping existing one-hop edges), which requires featurizing other node types?**  In the original paper and the tutorial, only metapaths that start with some type `A` and end with `A` are considered.  I think this is in fact comparing the metapath-based computation tree of each `A` node directly, and its expressiveness is limited.  To making a difference (thru backprop), more than one layers of HAN need to be used.  This might be similar to specifying all three-hop and four-hop metapaths though.
3. **Also, following the argument in 1, does it make sense to consider metapaths that don't necessarily have `A` on both ends (as the original paper did)?**  In this case, if we use more than one layer of HAN, considering metapaths would be conceptually similar to manually creating two layers.  Perhaps this is like we put some additional "focus" on `A`.
4. **Should there be any symmetries inserted into the `rev_X` edges?**  I understand that one needs to set up reverse edges for bidirectional message passing, but right now with `rev_X` it seems the weight matrices are reinitialized, while `X` and `rev_X` are symmetric relations in some sense.
5. **Is there a direct way to set up inductive training where the test `A` nodes would not be seen during training?**  I know there's built-in `train_mask`, but what I typically see in tutorials is that this is only used to select train nodes after message passing, where test nodes are involved.  I created a separate discussion question for this topic in #7269, where the simplest way I can think of is just to manually remove all edges attaching to the test nodes (while still keeping the nodes) in the KG before training for one time.
6. **I am also wondering if I can set up minibatch training using `NeighborLoader`** -- it looks like metapaths would be considered in the same way as normal edges so it should be fine, so long as the #iterations is processed in the same way for metapaths as normal edges.  (Since `HANConv` is quite memory costly on such a large graph, I would almost surely need to sample the graph.)

Please correct me if I am wrong anywhere!  Thanks in advance!

**Comment by rusty1s** [comment, marked as answer]
I am really sorry for super late reply :( 

1. I think this is dataset-specific. Certainly, metapaths are helpful but might not be capable of maintaining of all available information in your raw graph. That's why we provide the option in `AddMetaPaths` to keep or remove remaining information.
2. I think this makes sense, but is again dataset-specific. The goal of metapaths is mostly to reach important information faster/with less message passing steps.
3. Challenging question. You can certainly share weights across original and reverse edges. I doubt that this brings any model performance gain though. It gets a bit more unclear whether to do this if you think sharing weights for source and destination nodes of different types.
4. Yes, I think this is the correct way. You can just leverage `Data.subgraph()` method for this.
5. Yes, this should be no problem.
